### DISCUSSION TITLE
form does not submit because "x" is not a function

### DISCUSSION THREAD

**Discussion by jana-scheidemann** [neuefische/questions#827] [Web] [answered]
Hey,

I am currently working on my capstone project (DIY App - [GitHub-Link to project](https://github.com/jana-scheidemann/DIY-App)), which displays a list of all DIY-projects on the overview page (= `index.js`). Via the burger menu the user can add a new project. The link `Add New Project` opens the form (Component `ProjectFormNew` in the `ModalAdd`) to do so. The new project should be added to our list of DIY-projects = overview, and is supposed to show up on first position.

The code worked before, but since the app structure changed it does not anymore. No I get the error message: **Unhandled Runtime Error. TypeError: onAddProject is not a function**
```
components/ProjectFormNew.js (28:4) @ onAddProject

  26 |   };
  27 | 
> 28 |   onAddProject(newProject);
     |  ^
```

I am almost sure that I drilled the props correctly, I searched for typos, I console.log-ed the `newProject` (it gets the info it's supposed to get - at least in the `ProjectFormNew`) and I googled the error message, but I cannot figure it out.

[My `feature/refactor branch` is accessible via this link.](https://github.com/jana-scheidemann/DIY-App/tree/feature/refactor)

Thank you for your help!

**Comment by JessicaLoers** [comment, marked as answer]
**Hey @jana-scheidemann**  ✌️

This error is often related to the way props are handled or passed down through components, especially when you have nested components or a complex structure that has recently been refactored.

The component `Navigation`, which renders an instance of `ProjectFormNew`, was unintentionally used multiple times.  The error indicated that in at least one case the `Navigation` does not receive the function `onAddProject` as a prop before it is passed on to `ProjectFormNew`.

**Our solution**
1. Reduced Navigation Instances: By limiting the instances of the Navigation component to only those that are necessary, we've simplified the component structure.
2. Ensured Required Props: We've added the necessary properties, such as the `onAddProject` function, to the instances of the `Navigation` component that require it. This ensures that `ProjectFormNew` has access to these props and can function as intended. 🔥

**Reply by jana-scheidemann** [reply]
Thank you so much! It was a lot of work, but now it's fixed 🥳
